R/name_issues.R

```diff
@@ -1,74 +1,58 @@
-#' Table of GBIF name usage issues, with codes used in data output, full issue name,
-#' and descriptions.
+#' Parse and examine further GBIF name issues on a dataset.
 #'
-#' Table has the following fields:
+#' @export
 #'
-#' - code. Code for issue, making viewing data easier.
-#' - issue. Full name of the issue.
-#' - description. Description of the issue.
+#' @param .data Output from a call to [name_usage()], but only if
+#'   `return="all"`, or `return="data"`, otherwise function stops with error.
+#' @param ... Named parameters to only get back (e.g. bbmn), or to
+#' remove (e.g. -bbmn).
+#' @param mutate (character) One of:
 #'
-#' @export
-#' @usage name_issues()
-#' @seealso [gbif_issues()]
-#' @source <https://gbif.github.io/gbif-api/apidocs/org/gbif/api/vocabulary/NameUsageIssue.html>
-name_issues <- function() nameissues
+#' - `split` Split issues into new columns.
+#' - `expand` Expand issue abbreviated codes into descriptive names.
+#' for downloads datasets, this is not super useful since the
+#' issues come to you as expanded already.
+#' - `split_expand` Split into new columns, and expand issue names.
+#'
+#' For split and split_expand, values in cells become y ("yes") or n ("no")
+#'
+#' @references
+#' <https://gbif.github.io/gbif-api/apidocs/org/gbif/api/vocabulary/NameUsageIssue.html>
+#'
+#' @examples \dontrun{
+#' # what do issues mean, can print whole table
+#' head(gbif_issues())
+#' # or just name related issues
+#' gbif_issues()[which(gbifissues$type %in% c("name")),]
+#' # or search for matches
+#' gbif_issues()[gbif_issues()$code %in% c('bbmn','clasna','scina'),]
+#' # compare out data to after name_issues use
+#' (aa <- name_usage(name = "Lupus"))
+#' aa %>% name_issues("clasna")
+#'
+#' ## or parse issues in various ways
+#' ### remove data rows with certain issue classes
+#' aa %>% name_issues(-clasna, -scina)
+#'
+#' ### expand issues to more descriptive names
+#' aa %>% name_issues(mutate = "expand")
```

Comment:
I do understand this feature is not new and actually inspired by the existing `occ_issues` setup, but is this not misusing the `mutate` verb? I would expect `mutate` to be a function instead of an argument? Or I would intuitively expect a column with all values equal to `expand` interpreting this code without context ( where it actually is `mutate("expand")`).  Is this not making readability of the code harder? 

I would propose to change the argument name `mutate`

Reply:
I found it also strange as I come from `tidyverse` world. But, documentation was sufficiently good to unnderstand the difference. I didn't to change this as well as not part of my initial PR. @sckott: something to think about, but maybe not for this PR?

Reply:
the very immedate problem comes if you introduce a `rgbif::mutate` function that it clashes with `dplyr::mutate`, yes? is there a param name that is more clear than mutate?